### ISSUE TITLE
N805 classmethod-decorators x Pydantic

### ISSUE THREAD

**Issue by g-as** [charliermarsh/ruff#769]
Hi,

I have some **pydantic**'s models lying around with some validation like so:
```python
import pydantic


class MyModel(pydantic.BaseModel):

    my_field: str

    @pydantic.validator("my_field")
    def _lower(cls, my_field: str) -> str:
        ...
```

**pep8-naming** doesn't trigger with:
```
classmethod-decorators =
    classmethod
    validator
```
but **ruff** does.

The trick is `validator` is not a decorator but a function that returns a classmethod.

I also checked that, as expected, 
```python
from pydantic import validator


class MyModel(pydantic.BaseModel):

    my_field: str

    @validator
    def _lower(cls, my_field: str) -> str:
        ...
```
doesn't raise any **N805** error (even though it's not functional).

Other "issue", I could not get it to work when keeping the pydantic namespace:
```python
import pydantic


class MyModel(pydantic.BaseModel):

    my_field: str

    @pydantic.validator
    def _lower(cls, my_field: str) -> str:
        ...
```
even with
```toml
[tool.ruff.pep8-naming]
classmethod-decorators = [
    "classmethod",
    "validator",
    "pydantic.validator",
]
```

Is parity with pep8-naming on this attainable/desirable?

**Comment by charliermarsh** [comment]
Yeah that should work! Will take a look (but possibly tomorrow).

**Comment by g-as** [comment]
Man, you're on a mission.

Keep up the good work!

**Comment by charliermarsh** [comment]
Thank you :)

**Comment by charliermarsh** [comment]
Ok yeah the logic here just wasn't general enough! Will fix.

**Comment by charliermarsh** [comment]
Fixed and added that `pydantic` use-case as a dedicated test case. Will go out in the next release!

**Comment by charliermarsh** [comment]
Building now: https://github.com/charliermarsh/ruff/releases/tag/v0.0.124

**Comment by g-as** [comment]
cheers!
